database/src/backend/redb/env.rs

```diff
@@ -61,55 +67,124 @@ impl Env for ConcreteEnv {
             SyncMode::FastThenSafe | SyncMode::Threshold(_) => unimplemented!(),
         };
 
-        todo!()
-    }
-
-    #[cold]
-    #[inline(never)] // called once in [`Env::open`]?`
-    fn create_tables<T: Table>(&self, tx_rw: &mut Self::TxRw<'_>) -> Result<(), RuntimeError> {
-        todo!()
+        let env_builder = redb::Builder::new();
+
+        // TODO: we can set cache sizes with:
+        // env_builder.set_cache(bytes);
+
+        // Open the database file, create if needed.
+        let db_file = std::fs::OpenOptions::new()
+            .read(true)
+            .write(true)
+            .create(true)
+            .open(config.db_file())?;
+        let mut env = env_builder.create_file(db_file)?;
+
+        // Create all database tables.
+        // `redb` creates tables if they don't exist.
+        // <https://docs.rs/redb/latest/redb/struct.WriteTransaction.html#method.open_table>
+        use crate::tables::{TestTable, TestTable2};
+        let tx_rw = env.begin_write()?;
+
+        // FIXME:
+        // These wonderful fully qualified trait types are brought
+        // to you by `tower::discover::Discover>::Key` collisions.
+
+        // TODO: Create all tables when schema is done.
+
+        // TestTable
+        let table: redb::TableDefinition<
+            'static,
+            StorableRedb<<TestTable as Table>::Key>,
+            StorableRedb<<TestTable as Table>::Value>,
+        > = redb::TableDefinition::new(TestTable::NAME);
+        tx_rw.open_table(table)?;
+
+        // TestTable2
+        let table: redb::TableDefinition<
+            'static,
+            StorableRedb<<TestTable2 as Table>::Key>,
+            StorableRedb<<TestTable2 as Table>::Value>,
+        > = redb::TableDefinition::new(TestTable2::NAME);
+        tx_rw.open_table(table)?;
+
```

Comment:
same here with heed about seperating this into a function at some point so we don't have to repeate this for each table